test/runtests.jl

```diff
@@ -7,3 +7,89 @@ for prob in names(OptimizationProblems)
   prob_fn = eval(prob)
   prob_fn()
 end
+
+# Compare against CUTEst
+import CUTEst
+const problems_cutest = CUTEst.select()
+
+using NLPModels, JuMP, MathProgBase
+
+path = joinpath(dirname(@__DIR__), "src")
+files = filter(readdir(path)) do filename
+    endswith(filename, ".jl") && filename ∉ ("OptimizationProblems.jl", "template.jl")
+end
+
+struct Status
+    samenvars::Bool
+    samexval::Bool
+    sameobjval::Bool
+    samegradval::Bool
+end
+
+cutest_problem(name) = CUTEst.CUTEstModel(name)
+function optpr_problem(name, args...)
+    fmodel = getfield(OptimizationProblems, Symbol(name))
+    omodel = fmodel(args...)
+    x0 = getvalue.(omodel[:x])
+    d = JuMP.NLPEvaluator(omodel)
+    MathProgBase.initialize(d, [:Grad])
+    x0 = getvalue.(omodel[:x])
+    return d, x0
+end
+
+NLPModels.obj(d::JuMP.NLPEvaluator, x0) = MathProgBase.eval_f(d, x0)
+function NLPModels.grad(d::JuMP.NLPEvaluator, x0)
+    ∇f = similar(x0)
+    MathProgBase.eval_grad_f(d, ∇f, x0)
+    return ∇f
+end
+
+function compute_status(probname; showvals=false)
+    cmodel = cutest_problem(uppercase(probname))
+    d, x0 = optpr_problem(probname)
+    samenvars, samexval, sameobjval, samegradval = true, true, true, true
+    isok = false
+    if isa(x0, AbstractVector)
+        if length(x0) != length(cmodel.meta.x0)
+            samenvars = false
+            d, x0 = optpr_problem(probname, length(cmodel.meta.x0))
+        end
+        if (x0 ≈ cmodel.meta.x0)
+            samexval = false
+        end
+        showvals && @show length(x0)
+        if !(obj(cmodel, x0) ≈ obj(d, x0))
+            sameobjval = false
+            showvals && @show obj(cmodel, x0) obj(d, x0)
+        end
+        # Sometimes the variables are permuted. Allow for this possibility.
+        gc, go = sort!(grad(cmodel, x0)), sort!(grad(d, x0))
+        if !(gc ≈ go)
+            samegradval = false
+            c = gc \ go
```

Comment:
What's happening here?

Reply:
Trying to see if it's just linearly rescaled, e.g., if `f(x) -> c*f(x)` then the gradient components would be multiplied by a constant. I'll get rid of that.